updater/lib/dependabot/file_fetcher_job.rb

```diff
@@ -57,6 +57,10 @@ def clone_repo_contents
       return unless job.clone?
 
       file_fetcher.clone_repo_contents
+    rescue Dependabot::RepoNotFound, Dependabot::BranchNotFound
+      # base_commit_sha can't be reached, don't attempt to fetch it in the rescue
```

Comment:
I found this comment slightly confusing because we're inside a `rescue` clause ourselves, but it's easy enough to figure out you meant the `rescue` clause in the caller (`#perform`).

Having to reference other parts of the code in comments though makes me wonder if this needs a refactoring, so I was thinking of a more explicit approach, like this:

```diff
diff --git a/updater/lib/dependabot/file_fetcher_job.rb b/updater/lib/dependabot/file_fetcher_job.rb
index 415221ee5..58cb2c948 100644
--- a/updater/lib/dependabot/file_fetcher_job.rb
+++ b/updater/lib/dependabot/file_fetcher_job.rb
@@ -8,12 +8,15 @@ require "octokit"
 module Dependabot
   class FileFetcherJob < BaseJob
     def perform_job
+      @base_commit_sha = nil
+
       begin
         connectivity_check if ENV["ENABLE_CONNECTIVITY_CHECK"] == "1"
         clone_repo_contents
-        base_commit_sha
+        @base_commit_sha = file_fetcher.commit || "unknown"
         dependency_files
       rescue StandardError => e
+        @base_commit_sha ||= "unknown"
         if Octokit::RATE_LIMITED_ERRORS.include?(e.class)
           remaining = rate_limit_error_remaining(e)
           logger_error("Repository is rate limited, attempting to retry in " \
@@ -22,14 +25,14 @@ module Dependabot
           logger_error("Error during file fetching; aborting")
         end
         handle_file_fetcher_error(e)
-        service.mark_job_as_processed(job_id, base_commit_sha)
+        service.mark_job_as_processed(job_id, @base_commit_sha)
         clear_repo_contents_path
         return
       end
 
       File.write(Environment.output_path, JSON.dump(
                                             base64_dependency_files: base64_dependency_files.map(&:to_h),
-                                            base_commit_sha: base_commit_sha
+                                            base_commit_sha: @base_commit_sha
                                           ))
 
       save_job_details
@@ -40,7 +43,7 @@ module Dependabot
 
       File.write(Environment.job_path, JSON.dump(
                                          base64_dependency_files: base64_dependency_files.map(&:to_h),
-                                         base_commit_sha: base_commit_sha,
+                                         base_commit_sha: @base_commit_sha,
                                          job: job_definition["job"]
                                        ))
     end
@@ -57,10 +60,6 @@ module Dependabot
       return unless job.clone?
 
       file_fetcher.clone_repo_contents
-    rescue Dependabot::RepoNotFound, Dependabot::BranchNotFound
-      # base_commit_sha can't be reached, don't attempt to fetch it in the rescue
-      @base_commit_sha = "unknown"
-      raise
     end
 
     def base64_dependency_files
@@ -88,15 +87,6 @@ module Dependabot
       @job ||= Job.new(attrs)
     end
 
-    def base_commit_sha
-      @base_commit_sha ||= file_fetcher.commit || "unknown"
-    rescue StandardError
-      # If an error occurs, set the commit SHA instance variable (so that we
-      # don't raise when recording the error later) and re-raise
-      @base_commit_sha = "unknown"
-      raise
-    end
-
     def file_fetcher
       return @file_fetcher if defined? @file_fetcher
 
```